spec/lib/search_logic_spec.rb

```diff
@@ -0,0 +1,79 @@
+RSpec.describe SearchLogic do
+    it "saves new attractions" do
+        geocode_venice
+        
+        places_stub = stub_request(:get, venice_request_url("tourism.attraction")).to_return(status: 200, body: File.read("spec/responses/attractions_venice.json"))
+
+        expect {
+            result = SearchLogic.get_places_from_city_string("Venice, Italy", "attractions")
+            expect(result).to all(be_a(Attraction))
+        }.to change(City, :count).by(1).and change(Attraction, :count).by(20)
+
+        expect(geocode_venice).to have_been_requested.times(1)
```

Comment:
I'm surprised that this even works because it's calling `stub_request` twice. It'd be easier to follow if you assigned the stub and then made the assertion on it:

```suggestion
        geocode_stub = stub_geocode_venice
        
        places_stub = stub_request(:get, venice_request_url("tourism.attraction")).to_return(status: 200, body: File.read("spec/responses/attractions_venice.json"))

        expect {
            result = SearchLogic.get_places_from_city_string("Venice, Italy", "attractions")
            expect(result).to all(be_a(Attraction))
        }.to change(City, :count).by(1).and change(Attraction, :count).by(20)

        expect(geocode_stub).to have_been_requested.times(1)
```